pkg/uprobetracer/ldcache_parser.go

```diff
@@ -0,0 +1,199 @@
+package uprobetracer
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"strings"
+	"unsafe"
+)
+
+const (
+	cache1Header    string = "ld.so-1.7.0"
+	cache1HeaderLen int    = 11
+
+	cache2Header    string = "glibc-ld.so.cache"
+	cache2HeaderLen int    = 17
+	cache2Version   string = "1.1"
+
+	ldCache1EntrySize uint32 = 12
+	ldCache1Size      uint32 = 16
+	ldCache2EntrySize uint32 = 24
+	ldCache2Size      uint32 = 48
+)
+
+type ldCache1Entry struct {
+	Flags int32
+	Key   uint32
+	Value uint32
+}
+
+type ldCache1 struct {
+	Header     [11]int8
+	EntryCount uint32
+}
+
+type ldCache2Entry struct {
+	Flags int32
+	Key   uint32
+	Value uint32
+	Pad1_ uint32
+	Pad2_ uint64
+}
+
+type ldCache2 struct {
+	Header         [17]int8
+	Version        [3]int8
+	EntryCount     uint32
+	StringTableLen uint32
+	Pad_           [5]uint32
+}
+
+type ldEntry struct {
+	Key   string
+	Value string
+}
+
+func readFromBytes[T any](obj *T, rawData []byte) error {
+	if int(unsafe.Sizeof(*obj)) != len(rawData) {
+		return fmt.Errorf("reading from bytes: length mismatched")
+	}
+	buffer := bytes.NewBuffer(rawData)
+	err := binary.Read(buffer, binary.NativeEndian, obj)
```

Comment:
TIL, NativeEndian is available in Go 1.21 :)

Reply:
I'm using `NativeEndian` simply because I wasn't sure whether to use big-endian or little-endian. The C reference code I found just uses `read()`. Do you see any issues with this?

Reply:
No, I don't see any issues.